### DISCUSSION TITLE
What is the difference between transport and locator？

### DISCUSSION THREAD

**Discussion by ulongcha** [eProsima/Fast-DDS#3857] [Q&A]
When configure participants, transport and locator as well as physical port and logical port are confusing me. I can not configure them correctly. Can someone tell me what's the difference between them and how physical port and logical port work in discovery phase? If I want to use TCP how should I configure them correctly? Thank you very much!

**Comment by JesusPoderoso** [comment]
Hi @ulongcha , 
on one hand, the [transport](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/transport.html) layer is the responsible of sending and receiving messages. For instance, UDP, TCP or shared memory are transports.
On the other hand, a [locator](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/common.html#locatorlisttype) is an abstract concept that defines the physical properties of a transport. Those properties are IP address and port where communication is established.

In UDP transport, both physical and logical / virtual ports are the same, so Fast DDS only allows configuring one of them (as "port").
In TCP case, the port must be different, so both definitions are allowed (physical port and (logical) port options).
Please, check the [discovery](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery.html) section of the Fast DDS documentation for further information.

**Reply by ulongcha** [reply]
Hi @JesusPoderoso ,
Thank you for help! Can I consider that locator is like an configure item of transport and transport is responsible for the true data change turnnel? I also want to know how does logical port work in TCP case. Is there something notice when configuring  logical port? Can it be set randomly (available port lower than 65535) or should be calculated according to physical port?

**Reply by JesusPoderoso** [reply]
Hello @ulongcha, the [locator documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/transport_api.html#locator) has been improved recently, please consider reading it.

> Can I consider that locator is like an configure item of transport and transport is responsible for the true data change turnnel?

Yes, you are right!

> I also want to know how does logical port work in TCP case. 

In TCP, the distinction between logical and physical ports allows for several DDS applications using different RTPS ports (logical ports) to share the same physical port, thus only requiring for a single port to be opened for all communications
(opened to allow WAN traffic by port-forwarding your router for instance) .

> Is there something notice when configuring logical port? Can it be set randomly (available port lower than 65535) or should be calculated according to physical port?

The ports are [calculated by default](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/listening_locators.html#well-known-ports) based on the [OMG Standard](https://www.omg.org/spec/DDSI-RTPS/2.2/PDF) taking into account the domain ID and the participant ID, among other variables. 
If you want a custom port, it would be required to be higher than 49152 and lower than 65535.

**Reply by ulongcha** [reply]
Thank you very much!
